[Woo POS] [Cash & Receipts] UI updates

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #14808
https://github.com/woocommerce/woocommerce-ios/pull/14795 has to be merged first

<!-- Id number of the GitHub issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
This PR address several UI-related items regarding -mostly- cash & receipts
- Hide floating controls when we're performing cash operations: 94feda63f1d22aa29c9322dca36d5377b0f633f1
- Remove additional spacing at the bottom for cash payments view when it should display fullscreen bc041489f23dbf614f91082c49c44ed08c1876b6

- Removes differences in background colour on success view when payment has happened via cash: 1c21366f626ef76ff2800a54c02f89bc0626b90a A bit hard to see in screenshots, so I made it yellow just for debugging and clearly see the opacity differences:

| Before | After |
|--------|--------|
| <img width=600 src="https://github.com/user-attachments/assets/e5449902-29f5-4104-8fd7-80d37e22b5cb"> | <img width=600 src="https://github.com/user-attachments/assets/f9716a56-1bce-479d-a09f-955e6a81a89d"> | 

- Remove colour discrepancy in the payments view: 06ba4bc615ca28fb86567542b895319705f5ef19 at some point we ended having two different background colours here for the top and the bottom of the totals, when the latest designs show only one.

| Before | After |
|--------|--------|
| ![simulator_screenshot_F08ECD8A-8E6B-4180-987F-D8AA96377E19](https://github.com/user-attachments/assets/aa0fb07b-02e2-4be2-b508-757bb97c3426) |  ![20250109 background consistency on totals](https://github.com/user-attachments/assets/ee27bf82-cc90-46e1-8777-167e42ca4143) |

## Steps to reproduce
- Enable `.acceptCashForPointOfSale`
- Add items to the cart
- Observe that, before payment, we have a single background colour for the totals/subtotals/card reader view.
- Tap on "Cash Payment". Tap "Mark Payment as complete" (there's no validation merged yet, this is done on a [different PR](https://github.com/woocommerce/woocommerce-ios/pull/14811)
- Observe the success screen also has a single background colour.

**Feedback about transitions:**

I left transitions outside of this PR, but happy to tackle them as well. Ideally I think we're missing transition/animation effect between the cash payment view and order complete view, as well as between order complete view and email receipt view. I was expecting these to work automatically since we have transitions in `TotalsView` and `paymentView` but doesn't seem to be the case 🤔 

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
Only show POSFloatingControlView on card payment states

iamgabrielma committed:
Make content view full-screen by ignoring bottom safe area

iamgabrielma committed:
add cash success bg color case for consistency

iamgabrielma committed:
Remove color difference for payments view

iamgabrielma committed:
rename

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
remove unused associated value

jaclync reviewed (approved):
The mentioned changes tested as expected :shipit: 

Other observations:

- In dark mode, the cash input form has a border in a different background color.

<img src="https://github.com/user-attachments/assets/0bcca2ad-4756-4a14-bfba-c28ae1e0f6fd" width="500" />

- In the payment success screen, the content seems to be on the top side which looks more off when the device is in portrait. I'm not sure what is the expected design, as we focus on the landscape mode.

<img src="https://github.com/user-attachments/assets/ad5f987a-e795-424f-928d-cf496c31ec25" width="400" />

iamgabrielma committed:
Update payment collection button to latest design

iamgabrielma committed:
Update nav button formatted total alignment

iamgabrielma committed:
update bg color when collecting cash case

iamgabrielma committed:
return instance of payment view layout on cash payment case

breaking on this case was not enough to not apply default layout, but we needed to return a proper instance of the PaymentViewLayout

iamgabrielma commented:
Thanks for the review!

> In dark mode, the cash input form has a border in a different background colour.

Fixed https://github.com/woocommerce/woocommerce-ios/pull/14830/commits/5e63ef04478c30c2a104b9bcef8bee06423a0ae0

![Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2025-01-13 at 10 25 28](https://github.com/user-attachments/assets/4abe6dbd-212d-45fb-8192-bbc202ab94af)

> In the payment success screen, the content seems to be on the top side which looks more off when the device is in portrait. I'm not sure what is the expected design, as we focus on the landscape mode.

Good catch, addressed here: https://github.com/woocommerce/woocommerce-ios/pull/14830/commits/41061cbf62f433c41f679f7b409813c5161dc602

![Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2025-01-13 at 10 37 08](https://github.com/user-attachments/assets/d0faf7cb-94e8-4ba6-91d0-f70deac25505)

staskus reviewed (commented):
_ignore_

jaclync commented on WooCommerce/Classes/POS/Presentation/PointOfSaleCollectCashView.swift:
```diff
@@ -33,18 +33,22 @@ struct PointOfSaleCollectCashView: View {
+                                .font(.posBodyRegular)
+                                .foregroundColor(.primary)
+                        }
+                        .padding(.top, -Constants.navigationButtonSpacing)
```
super nit: the negative number could be in the constant

jaclync reviewed (approved):
Thanks for the updates, dark mode & portrait mode look good :shipit:
